### PR TITLE
Quickstart connection update

### PR DESCRIPTION
**Pull request by gemma-shay**
DOC-2040
Update quickstart options and connect doc to reflect new create sql user step and serverless connection modal

### CODE REVIEW

**Comment by gemma-shay** [comment]
@emily-horing @ericharmeling

**Review by emily-horing — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13525)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @emily-horing and @gemma-shay)

___
*[_includes/cockroachcloud/quickstart/create-first-sql-user.md, line 3 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-NyqDs-dT_DrtEB6S7:-N-NyqDs-dT_DrtEB6S8:b-98osk1) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/_includes/cockroachcloud/quickstart/create-first-sql-user.md#L3)):*
> ```Markdown
> The **Create SQL user** dialog allows you to create a new SQL user and password.
> 
> 1. Enter a username in the **SQL user** field.
> ```

.."or use the one provided by default"
___
*[cockroachcloud/connect-to-a-serverless-cluster.md, line 58 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-NzPztCcBA3kBklwUa:-N-NzPztCcBA3kBklwUb:b-kqxhn1) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/connect-to-a-serverless-cluster.md#L58)):*
> ```Markdown
>         <button class="filter-button page-level" data-scope="connection-parameters">Connection parameters</button>
>         <button class="filter-button page-level" data-scope="cockroachdb-client">CockroachDB client</button>
>         <button class="filter-button page-level" data-scope="python">Python</button>
> ```

Based on discussions with @kernfeld-cockroach we will probably release with just client, params, & general string to start. Others will be added as fast follow

**Review by ericharmeling — CHANGES_REQUESTED** [review]
Looking good! Please see my comments below.

> Reviewed 2 of 4 files at r1, 2 of 2 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13525#-:-N-O7gxO82E7aMmHeJ4x:b-hpm1hi)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gemma-shay and @kernfeld-cockroach)

___
*[_includes/cockroachcloud/create-a-sql-user.md, line 13 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-O2DbP9fry86908dvJ:-N-O2DbP9fry86908dvK:b6gyf1c) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/_includes/cockroachcloud/create-a-sql-user.md#L13)):*
> ```Markdown
> 1. Enter a username in the **SQL user** field.
> 1. Click **Generate & save password**.
> 1. Copy the **Generated password** and save it in a secure location.
> ```

It looks like `cockroachcloud/user-authorization.md` also needs for this information to be updated.



_Code quote:_
```Markdown
1. Click the **Add User** button in the top right corner.

    The **Create SQL user** dialog displays.

1. Enter a username in the **SQL user** field.
1. Click **Generate & save password**.
1. Copy the **Generated password** and save it in a secure location.
```
___
*[_includes/cockroachcloud/quickstart/get-connection-string.md, line 1 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-O2sA91dIZ__JBLLkA:-N-O2sA91dIZ__JBLLkB:b-r7px0b) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/_includes/cockroachcloud/quickstart/get-connection-string.md#L1)):*
> ```Markdown
> The **Connect to your cluster** dialog shows information about how to connect to your cluster.
> ```

nit: remove "your" here, since it's not in the title of the dialog.



_Suggestion:_
```Markdown
The **Connect to cluster** dialog shows
```
___
*[cockroachcloud/connect-to-a-serverless-cluster.md, line 22 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-NzGCeCw1bO34YT4sw:-N-NzGCeCw1bO34YT4sx:b-y42ocx) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/connect-to-a-serverless-cluster.md#L22)):*
> ```Markdown
> - [Create a Serverless (beta) cluster](create-a-serverless-cluster.html).
> - _(Optional)_ [Create a new SQL user](user-authorization.html#create-a-sql-user).
> - [Install CockroachDB](../stable/install-cockroachdb.html)
> ```

Do you need to install `cockroach` if you aren't using the SQL client/converting a connection string for JDBC?

Also - this is a step in the **Connect to cluster** dialog, if you select "CockroachDB Client".

IMO, you should remove this.
___
*[cockroachcloud/connect-to-a-serverless-cluster.md, line 30 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-O1ZpI8pCBXWZwZXhs:-N-O1ZpI8pCBXWZwZXht:b-r7px0b) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/connect-to-a-serverless-cluster.md#L30)):*
> ```Markdown
> 1. In the top right corner of the {{ site.data.products.db }} Console, click the **Connect** button.
> 
>     The **Connect to your cluster** dialog displays.
> ```

nit: remove "your" here, since it's not in the title of the dialog.



_Suggestion:_
```Markdown
**Connect to cluster** dialog displays
```
___
*[cockroachcloud/connect-to-a-serverless-cluster.md, line 38 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-O3ULlAk386ilxtYGa:-N-O3ULlAk386ilxtYGb:b-yotjzv) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/connect-to-a-serverless-cluster.md#L38)):*
> ```Markdown
> ## Step 2. Download the CA certificate
> 
> 1. In the **Download CA Cert** section of the dialog, select **Mac**, **Linux**, or **Windows** to adjust the commands used in the next steps accordingly.
> ```

IMO, we should remove all of this and have a single, imperative sentence, similar to what I've suggested below. This will make it easier for us to maintain the docs, and also will be much less confusing to the user, who should be copying and pasting from the CC console, and not from the docs site.



_Suggestion:_
```Markdown
In the **Download CA Cert** section of the dialog, select your operating system, and use the command provided to download the CA certificate to the default PostgreSQL certificate directory on your machine.
```
___
*[cockroachcloud/connect-to-a-serverless-cluster.md, line 50 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-O4Rch8pEp_19Bxkxx:-N-O4Rch8pEp_19Bxkxy:bcu7h8p) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/connect-to-a-serverless-cluster.md#L50)):*
> ```Markdown
>     {% include cockroachcloud/download-the-cert-free.md %}
>     
> ## Step 3. Connect to your cluster
> ```

nit: change the title to match the steps here (this will require updating references to this section too).



_Suggestion:_
```Markdown
Step 3. Get the connection string
```
___
*[cockroachcloud/connect-to-a-serverless-cluster.md, line 58 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-NzPztCcBA3kBklwUa:-N-O4D5D6yOKK-LsRdRd:b-5b77zl) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/connect-to-a-serverless-cluster.md#L58)):*
<details><summary><i>Previously, emily-horing wrote…</i></summary><blockquote>

Based on discussions with @kernfeld-cockroach we will probably release with just client, params, & general string to start. Others will be added as fast follow
</blockquote></details>

+1 to what Emily said. Let's remove all sections except the General connection string, Connection parameters, and CockroachDB client.
___
*[cockroachcloud/connect-to-a-serverless-cluster.md, line 92 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-O56X4CoFYqAcqFJu4:-N-O56X4CoFYqAcqFJu5:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/connect-to-a-serverless-cluster.md#L92)):*
> ```Markdown
>     Where:
>     - `<username>` is the SQL user. By default, this is your {{ site.data.products.db }} account username.
>     - `<password>` is the password for the SQL user. The password will be shown only once in the **Connection info** dialog after creating the cluster.
> ```





_Suggestion:_
```Markdown
The password will be shown only once in the **Connection info** dialog after creating the cluster, or in the **Create SQL user** dialog after creating the user.
```
___
*[cockroachcloud/connect-to-a-serverless-cluster.md, line 95 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-O5d8u8Cd_ybSuh8Kh:-N-O5d8v1M1S1bqD6pL9:b66lxz4) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/connect-to-a-serverless-cluster.md#L95)):*
> ```Markdown
>     - `<serverless-host>` is the hostname of the {{ site.data.products.serverless }} cluster.
>     - `<routing-id>` identifies your tenant cluster on a [multi-tenant host](architecture.html#architecture). For example, `funny-skunk-123`.
>     - `<cluster-id>` is a unique string used to identify your cluster when downloading the CA certificate. For example, `12a3bcde-4fa5-6789-1234-56bc7890d123`.
> ```

there is no `<cluster-id>` in the connection string above (just a `<routing-id>`).
___
*[cockroachcloud/connect-to-a-serverless-cluster.md, line 98 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-O602ACn39D52p2q-Z:-N-O602ACn39D52p2q-_:undefined) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/connect-to-a-serverless-cluster.md#L98)):*
> ```Markdown
>     You can find these settings by selecting the **Paramters only** option of the **Select option/language** dropdown.
> 
> 1. Add your copied connection string to your application code.
> ```





_Suggestion:_
```Markdown
1. Add your copied connection string to your application code. For information about connecting to {{ site.data.products.serverless }} with a [supported client](third-party-database-tools.html), see [Connect to a CockroachDB Cluster](connect-to-the-database.html).
```
___
*[cockroachcloud/quickstart.md, line 45 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-O6ltb8fVmxV4JB9OX:-N-O6ltb8fVmxV4JB9OY:b-r7px0b) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/quickstart.md#L45)):*
> ```Markdown
> <section class="filter-content" markdown="1" data-scope="python">
> 
> The **Connect to your cluster** dialog shows information about how to connect to your cluster.
> ```

nit: remove "your" here, since it's not in the title of the dialog.



_Suggestion:_
```Markdown
The **Connect to cluster** dialog
```
___
*[cockroachcloud/quickstart.md, line 71 at r3](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-NsZ-72aBfqonv_Nt1:-N-NsZ-72aBfqonv_Nt2:byn2iub) ([raw file](https://github.com/cockroachdb/docs/blob/384c49ca333a95ea124eb2d19a42d38e108e37c8/cockroachcloud/quickstart.md#L71)):*
> ```Markdown
> </section>
> 
> ## Step 3. Run the sample code
> ```

You added a new step 3, so this needs to be bumped to step 4.

**Comment by kernfeld-cockroach** [comment]
1. Can we consolidate the "quickstart" and "connect to a Serverless cluster?" They appear to overlap substantially. Perhaps out of the scope of this PR, though.
2. Maybe too spicy, but I think it's a red flag when documentation tells the user to click a specific button, because it should be clear in the UI.

**Comment by gemma-shay** [comment]
cc @ianjevans

**Comment by gemma-shay** [comment]
tftrs!

@kernfeld-cockroach the Quickstart uses the general connection string to run a node/python/java/go example app, the Connect doc should (ideally) be a more comprehensive description of all the ways to connect to a cluster. there's also substantial overlap with this doc that i would be more inclined to consolidate once all the new connect modal options are available: https://www.cockroachlabs.com/docs/stable/connect-to-the-database.html, but I think they all serve different purposes right now. Open to discussing more in future.

As to specifying buttons, we do that everywhere rn so that's a bigger docs conversation haha.

Does anyone have objects to removing the copy/pastable sql connection strings? IMO they became more of a nuisance than anything and it's not ideal for users to be filling in all the variables by hand when they can be copy/pasting from the modal.

Also... anyone know why the first check is failing?

**Comment by ericharmeling** [comment]
> Does anyone have objects to removing the copy/pastable sql connection strings?

Great idea. Those should just be in the console.

> Also... anyone know why the first check is failing?

This was happening to me last week when I tried updating some of the `cockroachcloud` include files. No idea why this is happening. @nickvigilante @jseldess Any ideas?

**Review by emily-horing — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13525)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @gemma-shay and @kernfeld-cockroach)

___
*[_includes/cockroachcloud/quickstart/get-connection-string.md, line 3 at r4](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-UadcmAVKIZ3BxYk42:-N-UadcmAVKIZ3BxYk43:b-n2ujnv) ([raw file](https://github.com/cockroachdb/docs/blob/d34e16936d57a850a5e8a74a9015fe2e4d8c8c3a/_includes/cockroachcloud/quickstart/get-connection-string.md#L3)):*
> ```Markdown
> The **Connect to your cluster** dialog shows information about how to connect to your cluster.
> 
> 1. Select **General connection string** from the **Select option/language** dropdown.
> ```

We should probably just call this the option dropdown since we won't be including the language snippets for now
___
*[_includes/cockroachcloud/quickstart/get-connection-string.md, line 4 at r4](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-UbvG30uQ4g0GiE1-K:-N-UbvG30uQ4g0GiE1-L:buejh3u) ([raw file](https://github.com/cockroachdb/docs/blob/d34e16936d57a850a5e8a74a9015fe2e4d8c8c3a/_includes/cockroachcloud/quickstart/get-connection-string.md#L4)):*
> ```Markdown
> 
> 1. Select **General connection string** from the **Select option/language** dropdown.
> 1. Copy the connection string displayed in the **General connection string** section and save it in a secure location.
> ```

We might want to say something like "Open the connection dropdown and copy the general connection string to save it in a secure location. This quickstart uses default certificates and you  may skip the certficate download instructions" for when the user selects javascript/java/go
then when python is selected we should note "download the certificate with the curl command displayed in the connection modal. Then copy the connection string.."

**Review by emily-horing — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13525)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @gemma-shay, and @kernfeld-cockroach)

___
*[_includes/cockroachcloud/quickstart/get-connection-string.md, line 6 at r5](https://reviewable.io/reviews/cockroachdb/docs/13525#-N-_yc5KDX3mrSFMPF7l:-N-_yc5KDX3mrSFMPF7m:bkf9t31) ([raw file](https://github.com/cockroachdb/docs/blob/e5fa5309b9e7d17830eb796d6ca2e489cff4d5d4/_includes/cockroachcloud/quickstart/get-connection-string.md#L6)):*
> ```Markdown
> 1. Open the **General connection string** section, then copy the connection string provided and save it in a secure location.
>     
>     This Quickstart uses default certificates, so you can skip the **Download CA Cert*** instructions.
> ```

minor nit, extra *

File: cockroachcloud/quickstart.md [hunk 1]
```diff
@@ -287,7 +283,7 @@ The **Connection info** dialog shows information about how to connect to your cl
 
     {% include_cached copy-clipboard.html %}
     ~~~ shell
-    export JDBC_DATABASE_URL=<jdbc-connection-string>
+    export JDBC_DATABASE_URL="<jdbc-connection-string>"
     ~~~
 
     The code sample uses the connection string stored in the environment variable `DATABASE_URL` to connect to your cluster.
```
Comment (ericharmeling) [inline cockroachcloud/quickstart.md:289]: Last-minute nit:

```suggestion
    The code sample uses the connection string stored in the environment variable `JDBC_DATABASE_URL` to connect to your cluster.
```

**Pull request merged**